app/services/worker_health_checker.rb

```diff
@@ -0,0 +1,70 @@
+# Helps with reading and writing queue health from Sidekiq
+module WorkerHealthChecker
+  module_function
+
+  # Reported directly to NewRelic when a queue appears unhealthy
+  class QueueHealthError < StandardError; end
+
+  # Sidekiq server-side middleware that wraps jobs and marks the queues as healthy
+  # when a job completes successfully
+  class Middleware
+    def call(_worker, _job, queue)
+      yield
+
+      WorkerHealthChecker.mark_healthy!(queue)
+    end
+  end
+
+  # Empty job that we put in each background queue to make sure the queue is running
+  # Relies on the Middleware to mark the queue as healthy
+  class DummyJob < ActiveJob::Base
+    def perform
+    end
+  end
+
+  # called on an interval to enqueues a dummy job in each queue
+  # @see deploy/schedule.rb
+  def enqueue_dummy_jobs
+    Sidekiq::Queue.all.each do |queue|
+      DummyJob.set(queue: queue.name).perform_later
+    end
```

Comment:
This is not tested according to Code Climate. I'm using their browser extension to see this. https://codeclimate.com/browser-extension